include/catch.hpp

```diff
@@ -292,6 +308,14 @@ using Catch::Detail::Approx;
 
 #define CATCH_ANON_TEST_CASE() INTERNAL_CATCH_TESTCASE_NO_REGISTRATION(INTERNAL_CATCH_UNIQUE_NAME( ____C_A_T_C_H____T_E_S_T____ ))
 
+#ifdef CATCH_PLATFORM_WINDOWS
```

Comment:
I assume that this is done to work around the non-conforming MSVC preprocessor. Because we support non-MSVC compilers on Windows, these changes should be guarded differently.

You might also want to incorporate a check for the new and fixed MSVC preprocessor:
https://blogs.msdn.microsoft.com/vcblog/2018/07/06/msvc-preprocessor-progress-towards-conformance/

Reply:
Yes it was done exactly for that reason :) Okay will try to limit this to something like
```c++
#if defined(_MSC_VER) && (!defined(_MSVC_TRADITIONAL) || (defined(_MSVC_TRADITIONAL) && _MSVC_TRADITIONAL)
//do the hacks
#else
//do the proper way
#endif
```